utils/src/tempfile.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::env::temp_dir;
+use std::path::Path;
+
+use vmm_sys_util::{errno::Result, tempfile};
+
+/// Wrapper over vmm_sys_util::tempfile::TempFile::new().
+/// This creates a temporary file inside $TMPDIR if set,
+/// otherwise inside `/tmp` if non-Android system.
+pub fn tempfile() -> Result<tempfile::TempFile> {
+    let mut in_tmp_dir = temp_dir();
+    in_tmp_dir.push("");
```

Comment:
why is this needed ?

Reply:
I'll leave a comment with the reason above the line in the code.

The reason is that the call for `temp_dir` is returning a `PathBuf` with the content `/tmp`. When creating a `TempFile::new(prefix)`, the `prefix` needs to be an explicit path/prefix where the temporary file name (a random string of 5 chars) will be placed/appended to.

If we would have just the path `/tmp`, when we will try to create a temporary file inside the `/tmp` directory, that will not happen because, the actual call to `TempFile::new("/tmp")` will create a file which will have a name similiar to this `/tmpXVFGB`, which is incorrect.

The call to `in_tmp_dir.push("")` will append a trailing `/`, which will result in `/tmp/` and which will make successful the call to `TempFile::new(in_tmp_dir)`.